internal/storage/v1/elasticsearch/spanstore/reader.go

```diff
@@ -151,6 +153,7 @@ func NewSpanReader(p SpanReaderParams) *SpanReader {
 		useReadWriteAliases: p.UseReadWriteAliases,
 		logger:              p.Logger,
 		tracer:              p.Tracer,
+		disableLegacyIds:    p.DisableLegacyIds,
```

Comment:
who not read the feature directly here, instead of passing down? you can still override it in tests